src/storage/src/region/tests/truncate.rs

```diff
@@ -0,0 +1,240 @@
+use crate::manifest::action::{RegionMetaAction, RegionMetaActionList, RegionTruncate};
+use crate::region::tests::{self, FileTesterBase};
+use crate::region::RegionImpl;
+use crate::test_util::config_util;
+use crate::test_util::flush_switch::{has_parquet_file, FlushSwitch};
+
+const REGION_NAME: &str = "region-truncate-0";
+
+/// Create a new region for truncate tests.
+async fn create_region_for_truncate(
+    store_dir: &str,
+    flush_strategy: FlushStrategyRef,
+) -> RegionImpl<RaftEngineLogStore> {
+    let metadata = tests::new_metadata(REGION_NAME);
+
+    let mut store_config =
+        config_util::new_store_config(REGION_NAME, store_dir, EngineConfig::default()).await;
+    store_config.flush_strategy = flush_strategy;
+
+    RegionImpl::create(metadata, store_config).await.unwrap()
+}
+
+/// Tester for truncate tests.
+struct TruncateTester {
+    store_dir: String,
+    base: Option<FileTesterBase>,
+}
+
+impl TruncateTester {
+    async fn new(store_dir: &str, flush_strategy: FlushStrategyRef) -> TruncateTester {
+        let region = create_region_for_truncate(store_dir, flush_strategy).await;
+        TruncateTester {
+            store_dir: store_dir.to_string(),
+            base: Some(FileTesterBase::with_region(region)),
+        }
+    }
+
+    #[inline]
+    fn base(&self) -> &FileTesterBase {
+        self.base.as_ref().unwrap()
+    }
+
+    async fn flush(&self) {
+        let ctx = FlushContext::default();
+        self.base().region.flush(&ctx).await.unwrap();
+    }
+
+    async fn truncate(&self) {
+        self.base().region.truncate().await.unwrap();
+    }
+
+    async fn reopen(&mut self) {
+        // Close the old region.
+        if let Some(base) = self.base.as_ref() {
+            base.close().await;
+        }
+        self.base = None;
+        // Reopen the region.
+        let store_config = config_util::new_store_config(
+            REGION_NAME,
+            &self.store_dir,
+            EngineConfig {
+                max_files_in_l0: usize::MAX,
+                ..Default::default()
+            },
+        )
+        .await;
+
+        let opts = OpenOptions::default();
+        let region = RegionImpl::open(REGION_NAME.to_string(), store_config, &opts)
+            .await
+            .unwrap()
+            .unwrap();
+
+        self.base = Some(FileTesterBase::with_region(region));
+    }
+}
+
+#[tokio::test]
+async fn test_truncate_basic() {
+    let dir = create_temp_dir("truncate-basic");
+    common_telemetry::init_default_ut_logging();
+    let store_dir = dir.path().to_str().unwrap();
+
+    let flush_switch = Arc::new(FlushSwitch::default());
+    let tester = TruncateTester::new(store_dir, flush_switch.clone()).await;
+
+    let data = [
+        (1000, Some("1000".to_string())),
+        (1001, Some("1001".to_string())),
+        (1002, Some("1002".to_string())),
+        (1003, Some("1003".to_string())),
+    ];
+
+    // Data in Memtable
+    let _ = tester.base().put(&data).await;
```

Comment:
nitpick(non-blocking): We have removed the lint on unused returned value so we don't need to ignore it explicitly on new code.
```suggestion
    tester.base().put(&data).await;
```